cmd/dlv/cmds/commands.go

```diff
@@ -470,9 +470,9 @@ func dapCmd(cmd *cobra.Command, args []string) {
 		} else { // reverse mode
 			headless = true // TODO(github.com/go-delve/delve/issues/2552): consider the same for the normal mode.
```

Comment:
This has been merged in. You can now factor out debug config outside of if-else.